lib/graphql/schema/warden.rb

```diff
@@ -106,6 +106,12 @@ def get_field(parent_type, field_name)
         @visible_parent_fields[parent_type][field_name]
       end
 
+      # @return [GraphQL::Argument, nil] The argument named `argument_name` on `parent_type`, if it exists
```

Comment:
```suggestion
      # @return [GraphQL::Argument, nil] The argument named `argument_name` on `parent_type`, if it exists and is visible
```

This would explain the `&&` logic